mmcls/models/backbones/vision_transformer.py

```diff
@@ -1,217 +1,79 @@
-        Returns:
-            Tensor: forwarded results with shape [num_query, bs, embed_dims].
-        """
-        query = x
-        if key is None:
-            key = query
-        if value is None:
-            value = key
-        if residual is None:
-            residual = x
-        if key_pos is None:
-            if query_pos is not None and key is not None:
-                if query_pos.shape == key.shape:
-                    key_pos = query_pos
-        if query_pos is not None:
-            query = query + query_pos
-        if key_pos is not None:
-            key = key + key_pos
-        out = self.attn(
-            query,
-            key,
-            value=value,
-            attn_mask=attn_mask,
-            key_padding_mask=key_padding_mask)[0]
-
-        return residual + self.dropout(out)
-
-
-class TransformerEncoderLayer(nn.Module):
+class TransformerEncoderLayer(BaseModule):
     """Implements one encoder layer in Vision Transformer.
 
     Args:
-        embed_dims (int): The feature dimension. Same as `FFN`.
-        num_heads (int): Parallel attention heads.
-        feedforward_channels (int): The hidden dimension for FFNs.
-        attn_drop (float): The drop out rate for attention layer.
-            Default 0.0.
-        proj_drop (float): Probability of an element to be zeroed
-            after the feed forward layer. Default 0.0.
-        act_cfg (dict): The activation config for FFNs. Defalut GELU.
+        embed_dims (int): The feature dimension
+        num_heads (int): Parallel attention heads
+        feedforward_channels (int): The hidden dimension for FFNs
+        drop_rate (float): Probability of an element to be zeroed
+            after the feed forward layer. Default 0.0
+        attn_drop_rate (float): The drop out rate for attention layer.
+            Default 0.0
+        drop_path_rate (float): stochastic depth rate. Default 0.0
+        num_fcs (int): The number of fully-connected layers for FFNs. Default 2
+        qkv_bias (bool): enable bias for qkv if True. Default True
+        act_cfg (dict): The activation config for FFNs. Defalut GELU
         norm_cfg (dict): Config dict for normalization layer. Default
-            layer normalization.
-        num_fcs (int): The number of fully-connected layers for FFNs.
-            Default 2.
+            layer normalization
+        batch_first (bool): Key, Query and Value are shape of
+            (batch, n, embed_dim)
+            or (n, batch, embed_dim). Default to False.
+        init_cfg (dict, optional): Initialization config dict
     """
 
     def __init__(self,
                  embed_dims,
                  num_heads,
                  feedforward_channels,
-                 attn_drop=0.,
-                 proj_drop=0.,
+                 drop_rate=0.,
+                 attn_drop_rate=0.,
+                 drop_path_rate=0.,
+                 num_fcs=2,
+                 qkv_bias=True,
```

Comment:
It seems that the default `qkv_bias` is `False` (`qkv_bias=False`) in most repos (e.g. [TIMM](https://github.com/rwightman/pytorch-image-models/blob/master/timm/models/vision_transformer.py#L136)), so maybe we can follow the common practices.

Reply:
In fact, the default value of qkv_bias is True. (e.g. [TIMM](https://github.com/rwightman/pytorch-image-models/blob/07fb05cc3d5c6f99d5b74a88e9a14db42e8e2e4f/timm/models/vision_transformer.py#L193))